src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1285,7 +1285,7 @@
     <value>'Stream' has a 'ReadAsync' overload that takes a 'Memory&lt;Byte&gt;' as the first argument, and a 'WriteAsync' overload that takes a 'ReadOnlyMemory&lt;Byte&gt;' as the first argument. Prefer calling the memory based overloads, which are more efficient.</value>
   </data>
   <data name="PreferStreamAsyncMemoryOverloadsMessage" xml:space="preserve">
-    <value>Change the '{0}' method call to the overload that receives a '{1}'.</value>
+    <value>Change the '{0}' method call to the overload '{1}'</value>
```

Comment:
@Youssef1313 the original message had the intention to let the user know that the `Stream` method currently being analyzed has an overload that takes a memory parameter, and we are suggesting the user to use that instead.

So if the original message was: `Change the '{0}' method call to the overload that receives a '{1}'.`, then the possibilities after formatting the string were:

- `Change the 'ReadAsync' method call to the overload that receives a 'Memory<byte>'.`
- `Change the 'WriteAsync' method call to the overload that receives a 'ReadOnlyMemory<byte>'.`

Your change makes the formatted message to be confusing. It will read:

- `Change the 'ReadAsync' method call to the overload 'Memory<byte>'.`
- `Change the 'WriteAsync' method call to the overload 'ReadOnlyMemory<byte>'.`

Can you please revert this change?

Reply:
@carlossanlop, did you see the issue this is fixing?

Reply:
Oh I see. Shouldn't we instead fix the second argument and make sure it's the name of the memory class being used by the overload?

Reply:
With the original string, the message would be much shorter.